lib/api/src/js/js_import_object.rs

```diff
@@ -0,0 +1,71 @@
+use crate::js::{Export, ExternType, Module, NamedResolver};
+use std::collections::HashMap;
+
+/// This struct is used in case you want to create an `Instance`
+/// of a `Module` with imports that are provided directly from
+/// Javascript with a JS Object.
+#[derive(Clone, Default)]
+pub struct JsImportObject {
+    module_imports: HashMap<(String, String), ExternType>,
+    object: js_sys::Object,
+}
+
+unsafe impl Send for JsImportObject {}
+unsafe impl Sync for JsImportObject {}
```

Comment:
JS objects are not Send/Sync, so this is technically unsound. However I believe this is only properly handled in #2646 and it might be acceptable to let this slide for now.

Reply:
I added a side comment to reflect this